### PR TITLE
[Testmerge pls] Dunc's proc define shit actually works, 85.6 -> 72.2 second startup

### PR DESCRIPTION
**Pull request by Iamgoofball**
# before
hg3lor6
# after
5yrwotx

@MrStonedOne @optimumtact @duncathan check 'em

ruin generation disabled to ensure no random results

my CPU is an intel i5-6600K at normal clock speeds

### CODE REVIEW

**Comment by duncathan** [comment]
activate your windows god damn

**Comment by Iamgoofball** [comment]
i'm too lazy to call microsoft since it was a rebuild and I'm using a new drive

**Comment by optimumtact** [comment]
I refuse to respect this result, also same map both times?

**Comment by Iamgoofball** [comment]
same map, yes, box-station

**Comment by PKPenguin321** [comment]
how the fuck is byond even real

**Comment by KorPhaeron** [comment]
@iamgoofball could you please try loading three 750 x 300 z levels (2 of lavaland pasted 3 times, one of space) before/after and see how much time it saves? I can do it tommorow if you dont feel like it though

**Comment by gbasood** [comment]
also possible that it loaded faster the second time because superfetch

**Comment by Thunder12345** [comment]
Why does this even work, it makes no sense

**Comment by Iamgoofball** [comment]
already checked gbasood

**Comment by Jalleo** [comment]
I request that others test this.

**Comment by xxalpha** [comment]
tgstation master (most recent)
````
This server is running a /tg/station 13 Git build.
You have unread updates in the changelog.Initialized Jobs subsystem within 0 seconds!
Initialized Ticker subsystem within 0 seconds!
Loading Box Station...
Initializing subsystems...
Initialized Jobs subsystem within 0 seconds!
Initialized Ticker subsystem within 0 seconds!
Unable to load next map config, defaulting to Box Station
Loading Box Station...
Please set up your character and select "Ready". The game will start in about 114.8 seconds.
Loaded station in 27.9s!
Loaded station in 27.9s!Initialized Mapping subsystem within 38.5 seconds!
Initialized Mapping subsystem within 38.5 seconds!Initialized Atoms subsystem within 32.5 seconds!
Initialized Machines subsystem within 0.1 seconds!
Initialized Atoms subsystem within 32.5 seconds!
Initialized Machines subsystem within 0.1 seconds!
Initialized Events subsystem within 0 seconds!
Initialized Events subsystem within 0 seconds!
Initialized Shuttle subsystem within 4.8 seconds!
Initialized Server Tasks subsystem within 0 seconds!
Initialized Shuttle subsystem within 4.8 seconds!
Initialized Server Tasks subsystem within 0 seconds!
Initialized Weather subsystem within 0 seconds!
Initialized Weather subsystem within 0 seconds!
HEY! LISTEN! 0.2 Seconds were wasted processing 69 turf(s) (connected to 3841 other turfs) with atmos differences at round start.
Initialized Air subsystem within 11.7 seconds!
Initialized Minimap subsystem within 0 seconds!
Initialized Assets subsystem within 0 seconds!
Initialized Air subsystem within 11.7 seconds!
Minimap generation disabled. Loading from cache...
Loaded cached minimap is outdated. There may be minor discrepancies in layout.
Initialized Minimap subsystem within 0 seconds!
Initialized Assets subsystem within 0 seconds!
Initialized Icon Smoothing subsystem within 6.7 seconds!
Initialized Icon Smoothing subsystem within 6.7 seconds!
Initialized Overlay subsystem within 7.6 seconds!
Initialized XKeyScore subsystem within 0 seconds!
Initialized Overlay subsystem within 7.6 seconds!
Initialized XKeyScore subsystem within 0 seconds!
Initialized Lighting subsystem within 19.3 seconds!
Initialized Persistence subsystem within 0 seconds!
Initializations complete within 122.1 seconds!
Initialized Lighting subsystem within 19.3 seconds!
Initialized Persistence subsystem within 0 seconds!
Initializations complete within 122.1 seconds!
Welcome to Psychic Carp Office Eight!
````

goof
````
This server is running a /tg/station 13 Git build.
You have unread updates in the changelog.Initialized Jobs subsystem within 0 seconds!
Initialized Ticker subsystem within 0 seconds!
Loading Box Station...
Initializing subsystems...
Initialized Jobs subsystem within 0 seconds!
Initialized Ticker subsystem within 0 seconds!
Unable to load next map config, defaulting to Box Station
Loading Box Station...
Please set up your character and select "Ready". The game will start in about 113.6 seconds.
Loaded station in 27.8s!
Loaded station in 27.8s!
Initialized Mapping subsystem within 39.7 seconds!
Initialized Mapping subsystem within 39.7 seconds!Initialized Atoms subsystem within 27.8 seconds!
Initialized Machines subsystem within 0.2 seconds!
Initialized Atoms subsystem within 27.8 seconds!
Initialized Machines subsystem within 0.2 seconds!
Initialized Events subsystem within 0 seconds!
Initialized Events subsystem within 0 seconds!
Initialized Shuttle subsystem within 4.9 seconds!
Initialized Shuttle subsystem within 4.9 seconds!
Initialized Server Tasks subsystem within 0 seconds!
Initialized Weather subsystem within 0 seconds!
Initialized Server Tasks subsystem within 0 seconds!
Initialized Weather subsystem within 0 seconds!
HEY! LISTEN! 0.1 Seconds were wasted processing 28 turf(s) (connected to 3777 other turfs) with atmos differences at round start.
Initialized Air subsystem within 11.2 seconds!
Initialized Minimap subsystem within 0 seconds!
Initialized Assets subsystem within 0 seconds!
Initialized Air subsystem within 11.2 seconds!
Minimap generation disabled. Loading from cache...
Loaded cached minimap is outdated. There may be minor discrepancies in layout.
Initialized Minimap subsystem within 0 seconds!
Initialized Assets subsystem within 0 seconds!
Initialized Icon Smoothing subsystem within 6.6 seconds!
Initialized Icon Smoothing subsystem within 6.6 seconds!
Initialized Overlay subsystem within 6.9 seconds!
Initialized XKeyScore subsystem within 0 seconds!
Initialized Overlay subsystem within 6.9 seconds!
Initialized XKeyScore subsystem within 0 seconds!
Initialized Lighting subsystem within 8.5 seconds!
Initialized Persistence subsystem within 0.1 seconds!
Initialized Lighting subsystem within 8.5 seconds!
Initialized Persistence subsystem within 0.1 seconds!
Initializations complete within 107 seconds!
Initializations complete within 107 seconds!
Welcome to North Fat Camp Eleven!
Tip of the round: You can spray a fire extinguisher or fire a gun while floating through space to change your direction. Simply fire opposite to where you want to go.
````

**Comment by Jalleo** [comment]
I believe this test confirms that it is a false positive.

**Comment by Razharas** [comment]
>He didnt run at least 10 tests to get statistical data

F

**Comment by xxalpha** [comment]
3 tgstation master tests: https://pastebin.com/sz3h5wVT
3 goofcode tests: https://pastebin.com/sWHVyx8q

**Comment by Jalleo** [comment]
Your lighting change is broken

File: code/modules/lighting/lighting_setup.dm [hunk 1]
```diff
@@ -1,8 +1,11 @@
 /proc/create_all_lighting_objects()
+/proc/create_lighting_objects_zlevel(zlevel)
+
+/create_all_lighting_objects()
```
Comment (xxalpha) [inline code/modules/lighting/lighting_setup.dm]: # GOOF THIS IS NOT HOW THIS LANGUAGE WORKS

Comment (Iamgoofball) [inline code/modules/lighting/lighting_setup.dm]: WHY DOES IT COMPILE

Comment (Iamgoofball) [inline code/modules/lighting/lighting_setup.dm]: WHY DOES IT COMPILE

Comment (xxalpha) [inline code/modules/lighting/lighting_setup.dm]: Good question, ask limmex.

**Comment by MrStonedOne** [comment]
you have to disable ruin generation using the config to proper test anything involving startup

**Comment by Iamgoofball** [comment]
They were shut off, see the OP

File: code/modules/lighting/lighting_setup.dm (hunk 1 above)
Comment (MrStonedOne) [inline code/modules/lighting/lighting_setup.dm]: I think this might actually be how this language works.

**Comment by Iamgoofball** [comment]
I tested this multiple times I'll test it again tomorrow it works

File: code/modules/lighting/lighting_setup.dm (hunk 1 above)
Comment (xxalpha) [inline code/modules/lighting/lighting_setup.dm]: The proc isn't called or breaks, as lighting_objects are never created, so it does not work.

**Comment by PKPenguin321** [comment]
>tomorrow

**Comment by optimumtact** [comment]
It doesn't work

**Pull request closed without merging**